In [2]:
from importlib import reload
import utils
reload(utils)
from utils import plots

import vgg16
reload(vgg16)
from vgg16 import Vgg16

Using Theano backend.


In [3]:
%matplotlib inline

In [7]:
path1 = "cancer_non_cancer_png/"
path2 = "benign_maligant_non_cancer_png/"

In [8]:
def classify(path):
    vgg = Vgg16()
    batch_size = 4
    batches = vgg.get_batches(path+'train', batch_size=batch_size)
    val_batches = vgg.get_batches(path+'valid', batch_size=batch_size*2)
    vgg.finetune(batches)
    vgg.fit(batches, val_batches, nb_epoch=1)

# Cancer Non Cancer
#trail 1 results
Epoch 1/1
56/56 [==============================] - 415s 7s/step - loss: 1.6457 - acc: 0.5580 - val_loss: 0.8953 - val_acc: 0.6146

#trail 2 results
Epoch 1/1
56/56 [==============================] - 388s 7s/step - loss: 1.4113 - acc: 0.5223 - val_loss: 1.3846 - val_acc: 0.6458


In [10]:
classify(path1)
classify(path2)

Found 226 images belonging to 3 classes.
Found 96 images belonging to 3 classes.
Epoch 1/1
56/56 [==============================] - 441s 8s/step - loss: 1.8231 - acc: 0.4688 - val_loss: 1.3078 - val_acc: 0.6354


# Benign Malignant Non Cancer
#trail 1 results
Epoch 1/1
56/56 [==============================] - 392s 7s/step - loss: 2.4346 - acc: 0.5000 - val_loss: 1.3231 - val_acc: 
0.6458

#trail 2 results
Epoch 1/1
56/56 [==============================] - 441s 8s/step - loss: 1.8231 - acc: 0.4688 - val_loss: 1.3078 - val_acc: 0.6354

